api/v1beta1/metal3cluster_types.go

```diff
@@ -66,6 +66,25 @@ type Metal3ClusterSpec struct {
 	// Default value is true, it is set in the webhook.
 	// +optional
 	CloudProviderEnabled *bool `json:"cloudProviderEnabled,omitempty"`
+
+	// FailureDomains specifies a list fo failure zones that can be used
+	// +optional
+	FailureDomains FailureDomains `json:"failureDomains,omitempty"`
+}
+
+// FailureDomains is a slice of FailureDomainSpecs.
+type FailureDomains map[string]FailureDomainSpec
+
+// FailureDomainSpec is the Schema for Cluster API failure domains.
+// It allows controllers to understand how many failure domains a cluster can optionally span across.
+type FailureDomainSpec struct {
```

Comment:
There is a struct for this already upstream. Can we reuse it instead of creating our own?
https://github.com/kubernetes-sigs/cluster-api/blob/4eb07516c7c3db623d798d08f7407eb44fc0c08d/api/core/v1beta2/cluster_types.go#L1614

Reply:
Here CAPI retrives failureDomains from infraCluster:
https://github.com/sbueringer/cluster-api/blob/3e9e41670af093243ce1b6ab8c1cd6be26ca5d5d/internal/controllers/cluster/cluster_controller_phases.go#L235

Reply:
I don't understand why we need v1beta2? 
CAPI FailureDomains struct is v1beta1. Check the following code
```
	FailureDomains clusterv1beta1.FailureDomains `json:"failureDomains,omitempty"`
```
https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/430d770b780e442b523c6a8d1747d3036614df32/apis/v1beta1/vspherecluster_types.go#L172